### PR TITLE
Added HTTP Client requests dump

### PR DESCRIPTION
**Pull request by AnandPilania**
- Added HTTP Client requests dump with 'http-client' type;
- 'table-v2' type for simple table view;

# 🛻 LaraDumps Pull Request

<br/><table><tr><td>❗ You <b>MUST</b> use this template to submit a Pull Request or it will NOT be accepted. ❗</td></tr></table><br/>

Welcome and thank you for your interest in contributing to our project.

## Guidelines

`   🗒️ `  Please read the [Contributing Guide](https://github.com/laradumps/laradumps/blob/main/CONTRIBUTING.md) and follow all steps listed there.

`   ✍️ `  Give this PR a meaningful title.

`   📣 `  Describe your PR details below.

### Motivation

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

### Related Issue(s)

This PR Fixes the Issue #_____.

### Description

This Pull Request adds dumping HTTP Client Requests with 'http-client' type + 'table-v2' type for support;

USE:
```
ds()->httpClientOn();
\Illuminate\Support\Facades\Http::get('https://jsonplaceholder.typicode.com/todos/1');
ds()->httpClientOff();
```

### Contribution Guide

- [x] I have read and followed the steps listed in the [Contributing Guide](https://github.com/laradumps/laradumps/blob/main/CONTRIBUTING.md).

### Documentation

 This PR requires [Documentation](https://github.com/laradumps/laradumps-docs) update?

- [x] Yes
- [ ] No
- [ ] I have already submitted a Documentation pull request.

### CODE REVIEW

File: src/Payloads/TableV2Payload.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    public function type(): string
+    {
+        return $this->type ?? 'table-v2';
+    }
+
+    public function content(): array
+    {
+        $values = array_map(function ($value) {
+            return $this->convertToPrimitive($value);
```
Comment (luanfreitasdev) [inline src/Payloads/TableV2Payload.php:37]: Please try to use the existing methods for this: Dumper::dump($value).

File: src/Payloads/TableV2Payload.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    /** @var array */
+    protected $values;
+
+    /** @var string */
+    protected $label;
+
+    /** @var null|string */
+    protected $type = null;
+
+    public function __construct(array $values, string $label = 'Table', string $type = 'table-v2')
```
Comment (luanfreitasdev) [inline src/Payloads/TableV2Payload.php:20]: Please use Constructor Promotion: https://www.php.net/manual/en/language.oop5.decon.php#language.oop5.decon.constructor.promotion

File: src/Observers/HttpClientObserver.php [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        if ($request->isMultipart()) {
+            return 'Multipart';
+        }
+
+        return 'Form';
+    }
+
+    protected function handleRequest(Request $request): Payload
+    {
+        return new TableV2Payload([
```
Comment (luanfreitasdev) [inline src/Observers/HttpClientObserver.php:85]: Can't use the current table? We need to define a single view type here. We will have to think about the future maintenance of this method

**Review by luanfreitasdev — CHANGES_REQUESTED** [review]

File: src/Payloads/TableV2Payload.php (hunk 2 above)
Comment (AnandPilania) [reply src/Payloads/TableV2Payload.php:20]: ohk... will convert it.

File: src/Observers/HttpClientObserver.php (hunk 3 above)
Comment (AnandPilania) [reply src/Observers/HttpClientObserver.php:85]: In current `table` there is `thead` but in this (`table-v2`), I only used `tbody` where first `td` acts as `head`;

I think its useful where we've to show data in `key => value` format;

Or may be you can suggest better option... [as I'm dumb in UI :)]

File: src/Payloads/TableV2Payload.php (hunk 1 above)
Comment (AnandPilania) [reply src/Payloads/TableV2Payload.php:37]: Will try in evening.

Comment (luanfreitasdev) [reply src/Payloads/TableV2Payload.php:26]: You can put the checks inside the method 👍

**Review by luanfreitasdev — APPROVED** [review]

**Pull request merged**